### PR TITLE
Switch to canImport instead of os conditions

### PR DESCRIPTION
**Pull request by FranzBusch**

### CODE REVIEW

**Review by tomerd — APPROVED** [review]

**Comment by Lukasa** [comment]
@FranzBusch Can you fix up the soundness stuff here?

File: Sources/Crypto/AEADs/AES/GCM/AES-GCM.swift [hunk 1]
```diff
@@ -11,10 +11,10 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 //===----------------------------------------------------------------------===//
-#if (os(macOS) || os(iOS) || os(watchOS) || os(tvOS)) && CRYPTO_IN_SWIFTPM && !CRYPTO_IN_SWIFTPM_FORCE_BUILD_API
+#if canImport(CryptoKit) && CRYPTO_IN_SWIFTPM && !CRYPTO_IN_SWIFTPM_FORCE_BUILD_API
 @_exported import CryptoKit
 #else
-#if (os(macOS) || os(iOS) || os(watchOS) || os(tvOS)) && !CRYPTO_IN_SWIFTPM_FORCE_BUILD_API
+#if canImport(CryptoKit) && !CRYPTO_IN_SWIFTPM_FORCE_BUILD_API
```
Comment (Lukasa) [inline Sources/Crypto/AEADs/AES/GCM/AES-GCM.swift:17]: This is wrong: these files are used to _build_ `CryptoKit`, so this branch can't safely rely on the `canImport` here. Prefer `canImport(Darwin)`.

**Review by Lukasa — CHANGES_REQUESTED** [review]

**Review by Lukasa — APPROVED** [review]
This LGTM.

**Comment by Lukasa** [comment]
Can you clean up the soundness and cmake things?

**Pull request merged**